test/bootstrap_fields_test.rb

```diff
@@ -59,20 +59,54 @@ class BootstrapFieldsTest < ActionView::TestCase
     expected = <<-HTML.strip_heredoc
       <div class="form-group">
         <label for="user_misc">Misc</label>
-        <input class="form-control-file" id="user_misc" name="user[misc]" type="file" />
+        <div class="custom-file">
+          <input class="custom-file-input" id="user_misc" name="user[misc]" type="file" />
+          <label class="custom-file-label" for="user_misc">Choose file</label>
+        </div>
       </div>
     HTML
     assert_equivalent_xml expected, @builder.file_field(:misc)
   end
 
+  test "file field placeholder can be customized" do
+    expected = <<-HTML.strip_heredoc
+      <div class="form-group">
+        <label for="user_misc">Misc</label>
+        <div class="custom-file">
+          <input class="custom-file-input" id="user_misc" name="user[misc]" type="file" />
+          <label class="custom-file-label" for="user_misc">Pick a file</label>
+        </div>
+      </div>
+    HTML
+    assert_equivalent_xml expected, @builder.file_field(:misc, placeholder: "Pick a file")
+  end
+
+  if ::Rails::VERSION::STRING > '5.1' && ::Rails::VERSION::STRING < '5.2'
```

Comment:
This looks good. If I really wanted to nit-pick, I would suggest changing the ID you use to be something other than what Rails != 5.1 would use, and then the test would be valid for all Rails versions >= 5.1. It would give coverage for that code path, since this helper doesn't use the standard `form_group_builder`/`form_group` approach.

Reply:
✅ done